packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1035,6 +1046,30 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
     }
   }
 
+  private getHighlightedIndexOnCharKeyPress = (keyPressed: string): number => {
+    const { highlightedIndex } = this.state
+    const filteredItemLowercasedStrings = this.getItemsFilteredBySearchQuery().map(item =>
```

Comment:
`getItemsFilteredBySearchQuery` can prove to be expensive for large `items` array and we're already calling it pretty often; can we add a `state` property, let's call it `itemsAsString` that keeps a copy of the items as string and update it only when `items` array changes?

```typescript
componentDidUpdate(prevProps: DropdownProps) {
  const items = this.props.items
  const prevItems = prevProps.items

  if (
    items.length !== prevItems.length ||
    _.difference(items, prevItems).length > 0 ||
    _.difference(prevItems, items).length > 0
  ) {
    this.trySetState({ itemsAsString: this.getItemsAsString() })
  }
}
```

changes should be minimal as `getItemsFilteredBySearchQuery` will use `itemsAsString` directly

Reply:
I also want to have the best performance, but only having these values in state won't help that much in my opinion. After you take a closer look in `getItemsFilteredBySearchQuery` you will see that the current implementation will not benefit much from the items as strings.

But maybe we can look into this a bit more and find a strategy that is better overall. Thanks!